policyengine_us/variables/gov/irs/income/taxable_income/adjusted_gross_income/irs_gross_income/social_security/tax_unit_taxable_social_security.py

```diff
@@ -25,8 +25,15 @@ def formula(tax_unit, period, parameters):
         )
         filing_status = tax_unit("filing_status", period)
 
-        base_amount = p.threshold.lower[filing_status]
-        adjusted_base_amount = p.threshold.upper[filing_status]
+        separate = filing_status == status.SEPARATE
+        cohabitating = tax_unit("cohabitating_spouses", period)
+        # Married filing separate who cohabitated has base amount and adj base amount of 0.
+        base_amount = p.threshold.lower[filing_status] * (
+            1 - separate * cohabitating
+        )
+        adjusted_base_amount = p.threshold.upper[filing_status] * (
+            1 - separate * cohabitating
+        )
```

Comment:
I think we should rather use a where statement here for clarity - both base and adjusted